lib/android-helpers.js

```diff
@@ -242,6 +242,13 @@ helpers.getLaunchInfo = async function (adb, opts) {
     return;
   }
   if (appPackage && appActivity) {
+    if (!new RegExp(/^[a-zA-Z0-9_\.]+$/).test(appPackage)) {
+      logger.warn(`appPackage, "${appPackage}", should be ^[a-zA-Z0-9_\.]+$`);
```

Comment:
I'd rather say `'${appPackage}' is expected to only include latin letters, digits, underscore and dot characters.`

Reply:
also, it might be handy to have a following note about where the first occurrence of an unexpected character has been detected, like

```
blabla$ <- at position 9
```

like compilers illustrate this